### ISSUE TITLE
filter_unhandled_paths broken with root_path set via Uvicorn

### ISSUE THREAD

**Issue by courtland** [stephenhillier/starlette_exporter#75]
I noticed that setting the `root_path` via Uvicorn breaks all metrics having a path label when `filter_unhandled_paths` is enabled.

The FastAPI docs [suggest that setting root_path via uvicorn is equivalent to the `FastAPI` attribute of the same name](https://fastapi.tiangolo.com/advanced/behind-a-proxy/#setting-the-root_path-in-the-fastapi-app)

This exporter however seems to make the assumption that it's set via FastAPI. I believe this is the problematic bit of code. 
https://github.com/stephenhillier/starlette_exporter/blob/0de22e78233cf88e746b68db76945965d0bfbedf/starlette_exporter/middleware.py#L375

In my case, I simply switched to setting `root_path` via the FastAPI attribute.

I'd be happy to submit a fix, but I'm not sure what the best way to inspect the uvicorn settings parameters in this context. Any tips would be appreciated.

Thanks!

**Comment by stephenhillier** [comment]
I don't know of a way to inspect uvicorn parameters from middleware, my guess is that it's not feasible.

I wonder if the only option is to add another optional config param `root_path` e.g.:
```
app.add_middleware(
  PrometheusMiddleware,
  app_name="hello_world",
  root_path="/api",
```

**Comment by courtland** [comment]
> I don't know of a way to inspect uvicorn parameters from middleware, my guess is that it's not feasible.
> 
> I wonder if the only option is to add another optional config param `root_path` e.g.:
> 
> ```
> app.add_middleware(
>   PrometheusMiddleware,
>   app_name="hello_world",
>   root_path="/api",
> ```

Yeah... I like that idea. Unfortunately I've noticed some weird differences (in FastAPI) between setting it via uvicorn and the FastAPI app.

**Comment by dolfinus** [comment]
This is probably caused by issue in Starlette, fixed only in 0.33.0: https://github.com/encode/starlette/pull/2352

But FastAPI does not support this version yet: https://github.com/tiangolo/fastapi/pull/10798
